app/Models/GeneralAssemblies/GeneralAssembly.php

```diff
@@ -147,7 +148,13 @@ public function isAttended(User $user): bool
      */
     public static function requirementsPassed(User $user): bool
     {
-        $lastAssemblies = GeneralAssembly::all()->sortByDesc('closed_at')->take(2);
+        $year_of_acceptance = $user->educationalInformation->year_of_acceptance;
+        $acceptance_date = now()->setYear($year_of_acceptance)->setMonth(9)->setDay(1);
+        $lastAssemblies = GeneralAssembly::orderBy('closed_at', 'desc')->take(2);
+        if ($lastAssemblies->count() >= 2 && $acceptance_date > $lastAssemblies->get()[1]->closed_at) {
```

Comment:
move second last assembly to a new var for better readability, then you don't even need the comment